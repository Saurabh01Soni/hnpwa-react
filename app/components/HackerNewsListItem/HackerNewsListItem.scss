.HackerNewsListItem {
  padding-left: 3em;
}

.HackerNewsListItem__title {
  text-decoration: none;
}

.HackerNewsListItem__info {
  font-size: 0.8em;
  margin-top: 0.3em;
}
