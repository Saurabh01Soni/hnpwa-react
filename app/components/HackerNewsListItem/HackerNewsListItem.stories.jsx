import React from 'react';
import { storiesOf } from '@storybook/react';
import { MemoryRouter } from 'react-router-dom';

import HackerNewsListItem from './';

const stories = storiesOf('HackerNewsListItem', module);

stories
  .addDecorator(story => (
    <MemoryRouter initialEntries={['/news/1']}>
      {story()}
    </MemoryRouter>
  ))
  .add('default', () => (
    <HackerNewsListItem
      commentsCount={48}
      id={15350263}
      timeAgo="20 hours ago"
      title="Keybase's mission is to make encryption mainstream"
      points={172}
      user="BradyDale"
      url="http://observer.com/2017/09/keybase-max-krohn-chris-coyne-okcupid/"
    />
  ));
