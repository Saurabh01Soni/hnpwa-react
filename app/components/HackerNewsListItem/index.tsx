import * as React from 'react';
import { Link } from 'react-router-dom';

import './HackerNewsListItem.scss';

export interface IHackerNewsListItemProps {
  readonly commentsCount?: number;
  readonly id?: number;
  readonly points?: number;
  readonly timeAgo?: string;
  readonly title?: string;
  readonly url?: string;
  readonly user?: string;
}

const defaultProps: IHackerNewsListItemProps = {
  commentsCount: 0,
  id: 0,
  points: 0,
  timeAgo: '',
  title: '',
  url: '',
  user: '',
};

const HackerNewsListItem: React.SFC<IHackerNewsListItemProps> = ({
  commentsCount,
  id,
  points,
  timeAgo,
  title,
  url,
  user,
}) => (
  <div className="HackerNewsListItem">
    <a className="HackerNewsListItem__title" href={url} target="__blank">
      {title}
    </a>
    <div className="HackerNewsListItem__info">
      {points !== null && `${points} points`}
      {user && ' by '}
      {user && (
        <Link className="HackerNewsListItem__link" href={`/user/${user}`} to={`/user/${user}`}>
          {user}
        </Link>
      )}
      {' '}
      {timeAgo} | {' '}
      <Link className="HackerNewsListItem__link" href={`/item/${id}/`} to={`/item/${id}`}>
        {commentsCount} comments
      </Link>
    </div>
  </div>
);

HackerNewsListItem.defaultProps = defaultProps;

export default HackerNewsListItem;
