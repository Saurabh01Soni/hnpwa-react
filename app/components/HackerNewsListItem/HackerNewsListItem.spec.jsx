import React from 'react';
import { shallow } from 'enzyme';

import HackerNewsListItem from './';

describe('<HackerNewsListItem />', () => {
  it('should match snapshot when render default', () => {
    const wrapper = shallow(<HackerNewsListItem />);
    expect(wrapper).toMatchSnapshot();
  });
});
