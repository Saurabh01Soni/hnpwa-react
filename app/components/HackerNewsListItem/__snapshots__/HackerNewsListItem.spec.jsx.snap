// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HackerNewsListItem /> should match snapshot when render default 1`] = `
<div
  className="HackerNewsListItem"
>
  <a
    className="HackerNewsListItem__title"
    href=""
    target="__blank"
  />
  <div
    className="HackerNewsListItem__info"
  >
    0
     points by 
     
    <Link
      replace={false}
      to="/user/"
    />
     
     | 
     
    <Link
      replace={false}
      to="/item/0"
    >
      0
       comments
    </Link>
  </div>
</div>
`;
